Close websocket when connection destroyed

DuBistKomisch opened the pull request:
Seems weird that there's no way to clean up after you don't need the cable any more. See e.g. https://github.com/algonauti/ember-cable/issues/12 and https://github.com/algonauti/ember-cable/issues/16.

This adds a simple method which stops the connection monitor polling to reconnect, and closes the websocket. Note that you still need to unsubscribe before calling this or you'll get the final disconnected event, which may or may not be a problem.

safeforge commented:
@DuBistKomisch can you rebase with latest master code?

DuBistKomisch commented:
Hmm, I don't think it's just a matter of rebasing, I'll have to rewrite it to work with the new code from that other PR. Will have a go tonight.

jasonmit commented:
Let me know if you need help with rebase also be sure to call super within the `destroy` or can move to `willDestroy` (UI won't let me leave a comment).

DuBistKomisch commented:
I actually didn't realise `destroy` was a built-in method, oops, I should definitely be using `willDestroy`. I'll try it out, thanks.

DuBistKomisch commented:
I'm pretty sure #29/#32 cover this, at least in terms of stopping the reconnect loop. @jasonmit do your changes explicitly close the websocket like this PR, or is this unnecessary anyway?

jasonmit commented:
@DuBistKomisch yup I believe you're correct.  As of my PR, if you invoke destroy on a consumer (& service itself) it disconnects everything and stops the polling.

DuBistKomisch commented:
Sorry guys, totally forgot about this. Rebased and removed the changes already covered in the other PRs. Boiled down to just explicitly closing the websocket while destroying everything, so I'll update the title too.
